### PR TITLE
feat: speed up sync with covalent

### PR DESCRIPTION
**Pull request by vadim-hleif**
now we are going to check for updates every 1s
and also ignore cases when there are no new blocks

### CODE REVIEW

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -33,7 +34,7 @@ contracts:
 network:
   chain-id: 80001
   covalent-api-key: 'ckey_103992e94cd94393beb35d1456d'
-  poll-interval: 'PT30S'
+  poll-interval: 'PT1S'
```
Comment (spoluyan) [inline src/main/resources/application.yaml:37]: You should also update this value in our infrastructure repo for each k8s deployment.

File: src/main/kotlin/nft/freeport/network/ContractsEventListenerHelper.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import org.eclipse.microprofile.context.ManagedExecutor
+import javax.enterprise.context.ApplicationScoped
+import javax.enterprise.event.Observes
+import javax.transaction.Transactional
+
+/**
+ * It's need to move triggering and initialization logic from [ContractsEventsListener] to write stable tests.
+ */
+@ApplicationScoped
+class ContractsEventListenerHelper(
```
[the remark this answers is not in the document]
Comment (vadim-hleif) [reply src/main/kotlin/nft/freeport/network/ContractsEventListenerHelper.kt]: done

File: src/main/kotlin/nft/freeport/network/ContractsEventsListener.kt [hunk 3]
```diff
@@ -74,26 +62,27 @@ class ContractsEventsListener(
...
 
-    @Scheduled(every = "{network.poll-interval}", skipExecutionIf = TestModeEnabled::class)
-    fun sync() {
-        contracts.map(ContractConfig::address)
-            .forEach { contract -> executor.execute { sync(contract) } }
-    }
+    fun sync(contract: String) {
+        val flag: AtomicBoolean = runningStatuses[contract] ?: error("$contract wasn't initialized")
+        // skip contract which is syncing
+        if (!flag.compareAndSet(false, true)) return
```
[the remark this answers is not in the document]
Comment (vadim-hleif) [reply src/main/kotlin/nft/freeport/network/ContractsEventsListener.kt:77]: yes, I just decided that it's more clear to use flag instead map api, but can fix it as you said.

yes, concurrent execution is disabled, but it's related only to the block on the screenshot below
image

according to concurrency: 
it guarantees that this function can't be launched twice, but there are no guarantees that executor will always handle all submitted lambdas before next triggering (each 1s in our case). So, it's better to have some protection against it. I can also log such a sase as warning (it shows that executor service so busy and the tasks delayed for some reasons, as far as I know we just use quarks executor which is used for processing all incoming requests).

File: src/main/kotlin/nft/freeport/network/ContractsEventsListener.kt [hunk 4]
```diff
@@ -52,17 +46,11 @@ class ContractsEventsListener(
 
     private val log = LoggerFactory.getLogger(javaClass)
 
-    private val contracts = contractsConfig.contracts().values
     private val lastScannedBlocks = ConcurrentHashMap<String, Long>()
-    private val runningStatuses = ConcurrentHashMap<String, Boolean>()
+    private val runningStatuses = ConcurrentHashMap<String, AtomicBoolean>()
```
[the remark this answers is not in the document]
Comment (vadim-hleif) [reply src/main/kotlin/nft/freeport/network/ContractsEventsListener.kt:50]: answered in the thread below 

just to have ability to change it when I get it without interaction with map. will change it to map api using if you wish

File: src/main/kotlin/nft/freeport/network/ContractsEventsListener.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (spoluyan) [reply src/main/kotlin/nft/freeport/network/ContractsEventsListener.kt:77]: So let's use `ConcurrentHashMap` with warning in log if we have concurrent executables.

[the remark this answers is not in the document]
Comment (vadim-hleif) [reply src/main/kotlin/nft/freeport/network/ContractsEventsListener.kt]: done

File: src/main/kotlin/nft/freeport/network/ContractsEventsListener.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (vadim-hleif) [reply src/main/kotlin/nft/freeport/network/ContractsEventsListener.kt]: done

**Review by spoluyan — APPROVED** [review]

**Pull request merged**